clippy_lints/src/types.rs

```diff
@@ -581,6 +609,24 @@ impl Types {
                             "a `VecDeque` might work",
                         );
                         return; // don't recurse into the type
+                    } else if let TyKind::Slice(s) = hir_ty.kind {
+                        if let TyKind::Path(ref qpath) = s.kind {
```

Comment:
This would probably be more readable using an `if_chain!`. See elsewhere in the code for how to use it.